docs/proposals/20230612-node-os-image-caching.md

```diff
@@ -0,0 +1,260 @@
+---
+title: Integrate Node OS Image caching with CAPZ
+authors:
+  - "@bennycortese"
+reviewers:
+  - "@nojnhuh"
+  - "@CecileRobertMichon"
+  - "@jackfrancis"
+  - "@willie-yao"
+creation-date: 2023-06-12
+last-updated: 2023-06-12
+
+status: provisional
+see-also:
+  - "https://en.wikipedia.org/wiki/Prototype_pattern"
+---
+
+# Title 
+Integrate Node OS Image caching with CAPZ
+
+## Table of Contents
+
+- [Integrate Node OS Image caching with CAPZ](#title)
+  - [Table of Contents](#table-of-contents)
+  - [Glossary](#glossary)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    - [Security Model](#security-model)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Alternatives](#alternatives)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+    - [Test Plan [optional]](#test-plan-optional)
+    - [Graduation Criteria [optional]](#graduation-criteria-optional)
+    - [Version Skew Strategy [optional]](#version-skew-strategy-optional)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+Node Prototype Pattern - Where we take a known good/working OS snapshot and make a “prototype” upon which all future node OS’ images are based.
+
+Warm nodes - The concept of keeping up extraneous and unused nodes to prevent having to wait for patches or security updates, thus allowing you to have those nodes ready when more users try to use your service without the wait times.
+
+Compute Image Gallery - A repository that is an Azure Resource for managing and sharing images
+
+Snapshot - A static copy of the image at the point in time that it is taken
+
+Prototype Node - Whichever node is chosen to clone the OS image of and cache it for future nodes to use
+
+## Summary
+
+The existing controllers will be modified to cache the Nodes’ OS image on a regular interval and update the model to use that image for future scale outs. This feature will:
+- allow for faster horizontal scaling 
```

Comment:
these bullet points are closer to motivation (they explain the "why"). Summary section should be a short 1-2 sentence summary of the feature proposed, or a "tldr" for the doc.

> The existing controllers will be modified

this is an implementation detail

Reply:
Fixed, thank you!